ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -211,6 +211,14 @@ export class AccountCreation extends React.Component<
   validate(): { [key: string]: string } {
     const { gsuiteDomain } = serverConfigStore.get().config;
 
+    // The validation data for this form is *almost* the raw Profile, except for the additional
+    // 'usernameWithEmail' field we're adding, to be able to separate our validation on the
+    // username itself from validation of the full email address. For this reason, we need to cast
+    // the profile object to 'any'.
+    const validationData = { ...this.state.profile } as any;
+    validationData.usernameWithEmail =
+      validationData.username + '@' + gsuiteDomain;
```

Comment:
Existing issue, but I think we can avoid the ugliness of the `as any` by just doing the following:

```suggestion
    // username itself from validation of the full email address.
    const validationData = {
      ...this.state.profile,
      usernameWithEmail: this.state.profile.username + '@' + gsuiteDomain
    };
```

Reply:
nice